### PR TITLE
approval-voting: populate session cache in advance

### PR DESCRIPTION
**Pull request by ordian**

### CODE REVIEW

**Review by rphmeier — APPROVED** [review]

**Comment by rphmeier** [comment]
Code changes look OK; this will just update the session window more aggressively.

**Review by drahnr — APPROVED** [review]
LGTM :+1: , given that the semantic change of `block's session` vs `block's child's session` is taken care of where it's used

**Comment by ordian** [comment]
bot merge

File: node/primitives/src/lib.rs [hunk 1]
```diff
@@ -69,7 +69,8 @@ pub const POV_BOMB_LIMIT: usize = (MAX_POV_SIZE * 4u32) as usize;
 /// On Polkadot this is 1 day, and on Kusama it's 6 hours.
 ///
 /// Number of sessions we want to consider in disputes.
-pub const DISPUTE_WINDOW: SessionIndex = 6;
+/// + 1 for the child's session.
+pub const DISPUTE_WINDOW: SessionIndex = 6 + 1;
```
Comment (rphmeier) [inline node/primitives/src/lib.rs:73]: I don't think this change is necessary.

1. BABE session changes occur at slots, so at particular points in time.
2. The parachains runtime defers all session changes by a single block in order to have predictability of the validator-set.
3.  Therefore, if we've imported one block which has `SessionIndexForChild(b) > SessionIndex(b)`, any blocks constructed by honest validators with accurate clocks after b will have `SessionIndexForChild(b') >= SessionIndexForChild(b)`.
4. What this means is that the change-set of being more aggressive with advancing the rolling session window will lead us to evict data at most 1 block-time (6 seconds) earlier than we could without the change. As this is dwarfed by the size of the window overall (hundreds or thousands of blocks), it doesn't justify a parameter tweak.

Comment (ordian) [reply node/primitives/src/lib.rs:73]: Agreed, but this param should be fetched from runtime anyway (#3227) and this change doesn't harm.

**Pull request merged**